Tests/RudderStackAnalyticsTests/SourceConfigModuleTests/SourceConfigProviderTests.swift

```diff
@@ -0,0 +1,355 @@
+        var receivedConfigs: [SourceConfig] = []
+        let cancellable = analytics.sourceConfigState.state
+            .sink { config in
+                receivedConfigs.append(config)
+            }
+        
+        defer { cancellable.cancel() }
+        
+        provider.fetchCachedConfigAndNotifyObservers()
+        
+        #expect(receivedConfigs.count == 1)
+        #expect(receivedConfigs.first?.source.sourceId.isEmpty == true)
+    }
+    
+    @Test("when fetching cached config with valid stored config, then notifies observers")
+    func testSourceConfigProvider_FetchCachedConfig_ValidConfig() async {
+        let storedConfig = SourceConfig(
+            source: RudderServerConfigSource(
+                sourceId: "cached-source-id",
+                sourceName: "Cached Source",
+                writeKey: "cached-write-key",
+                isSourceEnabled: true,
+                workspaceId: "cached-workspace",
+                updatedAt: "2023-10-24T10:00:00Z",
+                metricConfig: MetricsConfig(),
+                destinations: []
+            )
+        )
+        
+        mockStorage.write(value: storedConfig.jsonString, key: Constants.storageKeys.sourceConfig)
+        
+        var receivedConfigs: [SourceConfig] = []
+        let cancellable = analytics.sourceConfigState.state
+            .sink { config in
+                receivedConfigs.append(config)
+            }
+        
+        defer { cancellable.cancel() }
+        
+        provider.fetchCachedConfigAndNotifyObservers()
+        
+        #expect(receivedConfigs.count == 2) // Initial + cached
+        
+        guard let lastConfig = receivedConfigs.last else {
+            Issue.record("Can't read latest source configuration details")
+            return
+        }
+        
+        #expect(lastConfig.source.sourceId == "cached-source-id")
+        #expect(lastConfig.source.sourceName == "Cached Source")
+        #expect(lastConfig.source.writeKey == "cached-write-key")
+    }
+    
+    @Test("when fetching cached config with corrupted JSON, then does not notify observers")
+    func testSourceConfigProvider_FetchCachedConfig_CorruptedJSON() async {
+        
+        mockStorage.write(value: "invalid-json", key: Constants.storageKeys.sourceConfig)
+        
+        let provider = SourceConfigProvider(analytics: analytics)
+        
+        var receivedConfigs: [SourceConfig] = []
+        let cancellable = analytics.sourceConfigState.state
+            .sink { config in
+                receivedConfigs.append(config)
+            }
+        
+        defer { cancellable.cancel() }
+        
+        provider.fetchCachedConfigAndNotifyObservers()
+        
+        #expect(receivedConfigs.count == 1)
+        #expect(receivedConfigs.first?.source.sourceId.isEmpty == true)
+    }
+    
+    @Test("Given multiple observers, When SourceConfig is updated, Then all observers are notified")
```

Comment:
I think, we agreed to follow this small case format:

```suggestion
    @Test("given multiple observers, when SourceConfig is updated, then all observers are notified")
```

Can we ensure that we maintain this consistency in all the tests?

Reply:
Let's update this in other tests in this class.